utils/logalloc.cc

```diff
@@ -1372,17 +1374,17 @@ class region_impl final : public basic_region_impl {
 
     void free_segment(segment* seg, segment_descriptor& desc) noexcept {
         shard_segment_pool.free_segment(seg, desc);
-        if (_group) {
+        if (_listener) {
             _evictable_space -= segment_size;
-            _group->decrease_usage(_heap_handle, -segment::size);
+            _listener->decrease_usage(_heap_handle, -segment::size);
```

Comment:
It's equivalent change, but while we're at it -- why does this place call `decrease(-value)`? It would be less surprising if it  was ~~group~~ listener to apply minus to the value.

Reply:
It was like that since the beginning (e1eab5c84555d1492553911d8640260d4174de20). I can follow up with an improvement, but prefer to keep this a clean refactor.